controllers/nodemaintenance_controller.go

```diff
@@ -457,6 +458,7 @@ func (r *NodeMaintenanceReconciler) onReconcileErrorWithRequeue(ctx context.Cont
 		pendingList, _ := drainer.GetPodsForDeletion(nm.Spec.NodeName)
 		if pendingList != nil {
 			nm.Status.PendingPods = GetPodNameList(pendingList.Pods())
+			nm.Status.PendingPodsRefs = GetPodRefList(pendingList.Pods())
```

Comment:
At the end of the `Reconcile` function, `nm.Status.PendingPods` is set to `nil` in case of. I think we should do the same with this new `PendingPodsRefs`.